main.rb

```diff
@@ -0,0 +1,144 @@
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+
+    person_type = gets.chomp
+    if person_type != '1' && person_type != '2'
+      puts 'Invalid option'
+      return
+    end
+
+    print 'Age: '
+
+    age = gets.chomp
+
+    print 'Name: '
+
+    name = gets.chomp
+
+    person =
+      case person_type
+      when '1'
+        print 'Has parent permission? [Y/N]: '
+        # *If needed, ask for parameters for the option.
+        parent_permission = gets.chomp
+        parent_permission = parent_permission.downcase == 'y'
+
+        Student.new(age, name, parent_permission)
+      when '2'
+        print 'Specialization: '
+
+        specialization = gets.chomp
+
+        Teacher.new(age, specialization, name)
+      end
+
+    @people << person
+    puts 'Person created'
+  end
+
+  # *Create a book.
+  def create_book
+    print 'Title: '
+
+    title = gets.chomp
+
+    print 'Author: '
+
+    author = gets.chomp
+
+    @books << Book.new(title, author)
+    puts 'Book created'
+  end
+
+  # *Create a rental.
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    end
+
+    book_index = gets.chomp.to_i
+
+    puts "\nSelect a person from the following list by number (not id)"
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+
+    person_index = gets.chomp.to_i
+
+    print "\nDate: "
+
+    date = gets.chomp
+
+    @rentals << Rental.new(date, @books[book_index], @people[person_index])
+    puts 'Rental created successfully'
+  end
+
+  def list_rentals_for_person_id
+    print 'ID of person: '
+
+    id = gets.chomp.to_i
+
+    rentals = @rentals.filter { |rental| rental.person.id == id }
+    puts 'Rentals:'
+    rentals.each do |rental|
+      puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}"
+    end
+  end
+end
+
+def main
+  app = App.new
+
+  response = nil
+  puts "Welcome to School Library App!\n\n"
+  while response != '7'
+    puts 'Please choose an option by enterin a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+    response = gets.chomp
+    case response
+    when '1'
+      app.list_books
+    when '2'
+      app.list_people
+    when '3'
+      app.create_person
+    when '4'
+      app.create_book
+    when '5'
+      app.create_rental
+    when '6'
+      app.list_rentals_for_person_id
+    when '7'
+      puts 'Thank you for using this app!'
+    end
+    puts "\n"
+  end
+end
```

Comment:
- [ ] Your console app is not able to start on the command `ruby main.rb`. The _entry point_ is defined in the _main.rb_ but **not invoked**. Kindly ensure to invoke the _main_ method at the end of the _main.rb_ like this:
`main( )`. 
The specification of this project can be reviewed once this aspect is fixed and your console app starts correctly.

Reply:
Thanks, I did it and also fix a  problem  on students.rb